libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -108,71 +111,80 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		kubernetesAvailable: false,
 	}
 
-	client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
-	if err != nil {
-		if kubernetes.IsInCluster(config.KubeConfig) {
-			log.Debugf("Could not create kubernetes client using in_cluster config: %+v", err)
-		} else if config.KubeConfig == "" {
-			log.Debugf("Could not create kubernetes client using config: %v: %+v", os.Getenv("KUBECONFIG"), err)
-		} else {
-			log.Debugf("Could not create kubernetes client using config: %v: %+v", config.KubeConfig, err)
+	// complete processor's initialisation asynchronously so as to re-try on failing k8s client initialisations in case
+	// the k8s node is not yet ready.
+	go func(processor *kubernetesAnnotator) {
+		client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
+		if err != nil {
+			if kubernetes.IsInCluster(config.KubeConfig) {
+				log.Debugf("Could not create kubernetes client using in_cluster config: %+v", err)
+			} else if config.KubeConfig == "" {
+				log.Debugf("Could not create kubernetes client using config: %v: %+v", os.Getenv("KUBECONFIG"), err)
+			} else {
+				log.Debugf("Could not create kubernetes client using config: %v: %+v", config.KubeConfig, err)
+			}
+			return
 		}
-		return processor, nil
-	}
 
-	if !isKubernetesAvailable(client) {
-		return processor, nil
-	}
-
-	matchers := NewMatchers(config.Matchers)
+		connectionAttempts := 1
+		for ok := true; ok; ok = !isKubernetesAvailable(client) {
```

Comment:
`isKubernetesAvailable` is logging the result on every attempt, if kubernetes is not available it is going to log it 10 times at the info level. Should we only log it on the last attempt or something like this?